RAG_supporters/dataset/torch_dataset.py

```diff
@@ -41,6 +54,16 @@ class BaseDomainAssignDataset(Dataset):
 
         # Later load with DomainAssignDataset
         dataset = DomainAssignDataset(cache_dir='./cache')
+
+    Cluster Steering Usage:
+        builder = DomainAssignDatasetBuilder(
+            df=df,
+            embedding_model=embeddings,
+            steering_mode=SteeringMode.CLUSTER_DESCRIPTOR,
+            cluster_labels=cluster_assignments,  # Dict[int, List[int]] or soft labels
+            cluster_descriptors=topic_descriptors,  # Dict[int, List[str]]
+            return_triplets=True  # Returns (base, steering, target)
+        )
     """
 
     def __init__(
```

Comment:
Add function for validation:
- Dataset is non-empty
- DataFrame has required columns
- Embedding model works
- Cluster IDs are valid
- Embeddings have consistent dimensions